api-tester/src/main/java/com/revenuecat/apitester/java/EntitlementInfoAPI.java

```diff
@@ -0,0 +1,55 @@
+package com.revenuecat.apitester.java;
+
+import com.revenuecat.purchases.EntitlementInfo;
+import com.revenuecat.purchases.OwnershipType;
+import com.revenuecat.purchases.PeriodType;
+import com.revenuecat.purchases.Store;
+
+import java.util.Date;
+
+@SuppressWarnings({"unused"})
+final class EntitlementInfoAPI {
+    static void check(final EntitlementInfo entitlementInfo) {
+        final String identifier = entitlementInfo.getIdentifier();
+        final boolean active = entitlementInfo.isActive();
+        final boolean willRenew = entitlementInfo.getWillRenew();
+        final PeriodType periodType = entitlementInfo.getPeriodType();
+        final Date latestPurchaseDate = entitlementInfo.getLatestPurchaseDate();
+        final Date originalPurchaseDate = entitlementInfo.getOriginalPurchaseDate();
+        final Date expirationDate = entitlementInfo.getExpirationDate();
+        final Store store = entitlementInfo.getStore();
+        final String productIdentifier = entitlementInfo.getProductIdentifier();
+        final boolean sandbox = entitlementInfo.isSandbox();
+        final Date unsubscribeDetectedAt = entitlementInfo.getUnsubscribeDetectedAt();
+        final Date billingIssueDetectedAt = entitlementInfo.getBillingIssueDetectedAt();
+        final OwnershipType ownershipType = entitlementInfo.getOwnershipType();
+    }
+
+    static void store(final Store store) {
+        switch (store) {
+            case APP_STORE:
+            case MAC_APP_STORE:
+            case PLAY_STORE:
+            case STRIPE:
+            case PROMOTIONAL:
+            case UNKNOWN_STORE:
+            case AMAZON:
```

Comment:
i wonder if we want to somehow fail on `else`, so that we remember to add any new values?

Reply:
Nope, but fixed, thank you!!
I learned something new: https://proandroiddev.com/til-when-is-when-exhaustive-31d69f630a8b

I like that this can be handled in Kotlin better than in Java :)

<img width="690" alt="Screen Shot 2021-12-28 at 17 39 59" src="https://user-images.githubusercontent.com/685609/147619422-b44b15a8-f7b8-4c9f-b389-ba1514e74629.png">

Reply:
Same for Java:
<img width="651" alt="Screen Shot 2021-12-28 at 17 55 49" src="https://user-images.githubusercontent.com/685609/147620106-61f794e6-eb2b-4bf8-9322-abcc0432922e.png">

Reply:
Actually we can't do it for Java because this requires Java 14.